geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -237,15 +238,21 @@ public int hstrlen(byte[] field) {
     return new ArrayList<>(hash.keySet());
   }
 
-  public ImmutablePair<Integer, List<byte[]>> hscan(Pattern matchPattern,
-      int count,
-      int cursor) {
-
-    ArrayList<byte[]> resultList = new ArrayList<>(count + 2);
+  public ImmutablePair<Integer, List<byte[]>> hscan(Pattern matchPattern, int count, int cursor) {
+    // No need to allocate more space than it's possible to use given the size of the hash. We need
+    // to add 1 to hash.size() to ensure that if count > hash.size(), we return a cursor of 0
+    long maximumCapacity = 2L * Math.min(count, hlen() + 1);
+    if (maximumCapacity > Integer.MAX_VALUE) {
+      LogService.getLogger().error(
```

Comment:
I'm not sure what we should do about logging in these cases. Normally we do not log something in a server when a client does something wrong. We just have the server respond to the client with an error message. I know in some cases we are limited by the redis standard error responses in giving the user extra information about what they did wrong. In that case we could add some info in the server log but maybe it would be at the "info" log level. The "error" level is usually reserved for when something unexpected happened on the server but it thinks it can keep running. The "warning" level is if things don't look healthy (but it might be okay) and we want to get the admin's attention. The "fatal" level is if something is wrong and the server needs to shutdown.
In this case I would not log anything here if we can tell the client what was wrong. If we can't get a response back to the client then I think this should be "info" and describe what the client did wrong.

Reply:
I've changed the log level to info and made the message more descriptive/helpful.